### ISSUE TITLE
CI: build SIFDecode and CUTEst

### ISSUE THREAD

**Issue by dpo** [ralna/ARCHDefs#32]
Updates to the compilers here can, and do, cause breakage of the Homebrew formulas for SIFDecode and CUTEst. CI should build them from source for macOS and Linux to make sure the selection mechanism and compilation go through.

**Comment by nimgould** [comment]
Agreed. But archdefs is a relic from the makefiles days, and has always been static until someone updates the relevant preference file. Of course, the makefile install always prompts the installer to check the compiler-specific data before committing herself, since make doesn't know what system components are available. The archdefs files are only a suggestion which might work.

**Comment by amontoison** [comment]
@dpo we now have a Meson build system for the four packages (SPRAL, SIFDecode, GALAHAD, CUTEst).

Precompiled binaries are also automatically generated when we tag a new release with the terminal.
```
git tag v5.0.0
git push --tags
```

**Comment by dpo** [comment]
It's great news that ARCHDefs will no longer be necessary thanks to Meson. I'll carve out some time to update the Homebrew recipes.
